plugins/woocommerce/client/legacy/js/admin/wc-shipping-zone-methods.js

```diff
@@ -351,6 +351,44 @@
 
 					$( document.body ).trigger( 'init_tooltips' );
 				},
+				// Cost values need to be stripped of thier thousandth separators and made sure
+				// the decimal separator is a ".".
+				unformatShippingMethodNumericValues: function( data ) {
+					if ( ! window.wc.ShippingCurrencyContext ) {
+						return data;
+					}
+
+					const config = window.wc.ShippingCurrencyContext.getCurrencyConfig();
+					const numericValuesFields = [
+						'woocommerce_free_shipping_min_amount',
+						'woocommerce_flat_rate_cost',
+					];
+
+					numericValuesFields.forEach( function( field ) {
+						const formattedValue = data[ field ];
+						
+						if ( ! formattedValue ) {
+							return;
+						}
+
+						if ( 'number' === typeof formattedValue ) {
+							return;
+						}
+
+						// Brackets signal a formula. Avoid unformatting these values.
```

Comment:
Could we maybe have some logic to sanitize the string, split it into components (number/not number), and then run the sanitization on all numbers instead of not doing anything at all when there's a shortcode?

Reply:
I agree it's outside the scope of this PR - and possibly entirely outside the scope of what we should support. There was a previous PR that tried to allow this but it got reverted due to the mess it made.

I do think we should define a grammar for the formula and then parse it according to the grammar though, that shouldn't be too much work and will be reuseable going forward.

Reply:
(with just standard ISO decimal numbers in the formula)